### PR TITLE
A storage for the research department

### PR DESCRIPTION
**Pull request by FieryTech**
proposition

A pretty simple addition:
- 5x7 room with scientist access attached to the airlock.
- A nice showcase, the windoor is locked to RD only.
- A gun cabinet with scientist access, for the prototypes.
- Four crate emplacement, for that sweet storage space.
- A exosuit charging pad, where you can hide your combat exosuit.
- The Research Director can lock the room, by pressing the button.
- Camera, air alarm, APC, etc. STUFF, BUTTONS! Everything.

It's my first contribution ever, please be gentle. :b

### CODE REVIEW

**Comment by NonQueueingMatt** [comment]
I think the gun cabinet should have different colors (science colors, specifically) to not be confusing.

File: maps/aurora/aurora-4_mainlevel.dmm [hunk 1]
```diff
@@ -6562,14 +6589,13 @@
...
-	icon_state = "pipe-c"
+/obj/effect/floor_decal/corner_wide/mauve/full{
+	dir = 8
 	},
-/obj/effect/floor_decal/corner_wide/mauve{
-	dir = 9
+/obj/machinery/vending/snack{
+	step_x = 0;
+	step_y = 0
 	},
```
Comment (Amunak) [inline maps/aurora/aurora-4_mainlevel.dmm:6598]: I believe this should never appear anywhere.
```suggestion
/obj/machinery/vending/snack,
```

File: maps/aurora/aurora-4_mainlevel.dmm [hunk 2]
```diff
@@ -9799,6 +9888,72 @@
...
+	id = "RDStorage";
+	name = "Storage lockdown";
+	opacity = 0
+	},
+/obj/machinery/button/remote/blast_door{
+	id = "RDStorage";
+	name = "Storage lockdown";
+	req_access = list(30);
+	step_x = 5;
+	step_y = 27
```
Comment (Amunak) [inline maps/aurora/aurora-4_mainlevel.dmm:9951]: Same here.
```suggestion
	req_access = list(30)
```

**Review by Amunak — CHANGES_REQUESTED** [review]

**Comment by FieryTech** [comment]
> 
> 
> I think the gun cabinet should have different colors (science colors, specifically) to not be confusing.

I asked a friend to recolor the sprite, will implement on reception!

**Comment by fernerr** [comment]
Very well done on your first contribution, props. One thing, rather than having the area be the /area/rnd/lab with the name changed via a manual edit, I'd prefer you define a new area for this room somewhere here: https://github.com/Aurorastation/Aurora.3/blob/a9514e19ed6ce1a854de6aa1602ba871f423fa5c/maps/_common/areas/station/research.dm#L28 
As /area/rnd/storage is already used by toxins, another path name needs to be chosen, perhaps /area/rnd/strongroom

**Comment by FieryTech** [comment]
proposition
- Addition of a Science gun locker with sprites and script, science access lock.
- Addition of a new static area for the storage, and not a manual one.

**Comment by FieryTech** [comment]
Reintroduced Hydrogen and Holopad PR to resolve version conflicts.

!review

**Comment by mikomyazaki** [comment]
Fixed some problems with commit history at author's request.

**Review by fernerr — APPROVED** [review]

**Comment by Amunak** [comment]
Somehow the diff still shows ~~some holopads as removed~~ the science holopad as moved? 🤔

**Comment by FieryTech** [comment]
Should be fixed, the diff is now perfectly equal to actual state, and shows this PR changes.

**Comment by FieryTech** [comment]
Fixed, the holopad was one meter too far to the right compared to the current server map.

**Review by Geevies — APPROVED** [review]

**Review by Amunak — APPROVED** [review]

**Pull request merged**